clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DisableOpenstackAtomicOperation.groovy

```diff
@@ -36,23 +39,45 @@ class DisableOpenstackAtomicOperation extends AbstractEnableDisableOpenstackAtom
   }
 
   @Override
-  Void addOrRemoveInstancesFromLoadBalancer(List<String> instanceIds, List<String> loadBalancers) {
+  Void addOrRemoveInstancesFromLoadBalancer(List<String> instanceIds, List<String> loadBalancerIds) {
     task.updateStatus phaseName, "Deregistering instances from load balancers..."
+    Map<String, Future<LoadBalancerV2StatusTree>> statusTrees = [:]
+    Map<String, Future<String>> ips = instanceIds.collectEntries { instanceId ->
+      task.updateStatus phaseName, "Getting ip for instance $instanceId..."
+      [(instanceId): CompletableFuture.supplyAsync({
+        provider.getIpForInstance(description.region, instanceId)
+      } as Supplier<String>).exceptionally { t ->
```

Comment:
I can't exactly log the task status inline, as the thread has no access to the task.

I will add logging when the result is retrieved and not found.